single_include/inja/inja.hpp

```diff
@@ -2060,7 +2061,7 @@ class Parser {
       : config(parser_config), lexer(lexer_config), template_storage(template_storage), function_storage(function_storage) {}
 
   Template parse(std::string_view input, std::string_view path) {
-    auto result = Template(static_cast<std::string>(input));
+    auto& result = template_stash.emplace_back(Template(static_cast<std::string>(input)));
```

Comment:
To me it seems like it might be more correct to use `make_shared` to handle this